src/java.base/share/classes/sun/security/ssl/SSLBasicKeyDerivation.java

```diff
@@ -28,29 +28,33 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.security.GeneralSecurityException;
-import java.security.spec.AlgorithmParameterSpec;
+import javax.crypto.KDF;
 import javax.crypto.SecretKey;
+import javax.crypto.spec.HKDFParameterSpec;
 import javax.net.ssl.SSLHandshakeException;
 
+import sun.security.ssl.CipherSuite.HashAlg;
+
 final class SSLBasicKeyDerivation implements SSLKeyDerivation {
-    private final String hashAlg;
+    private final String hkdfAlg;
     private final SecretKey secret;
     private final byte[] hkdfInfo;
+    private final int keyLen;
 
-    SSLBasicKeyDerivation(SecretKey secret, String hashAlg,
-            byte[] label, byte[] context, int length) {
-        this.hashAlg = hashAlg.replace("-", "");
+    SSLBasicKeyDerivation(SecretKey secret, HashAlg hashAlg, byte[] label,
+            byte[] context) {
+        this.hkdfAlg = hashAlg.hkdfAlgorithm;
         this.secret = secret;
-        this.hkdfInfo = createHkdfInfo(label, context, length);
+        this.hkdfInfo = createHkdfInfo(label, context, hashAlg.hashLength);
+        this.keyLen = hashAlg.hashLength;
```

Comment:
Very minor nit: might be worth accessing this field once and passing `this.keyLen` to `createHkdfInfo` instead.

Reply:
Hmm, I like to match the ordering in the constructor with the ordering of fields. Since this is relatively minor, I am inclined to leave it as is...